qcodes/instrument/parameter.py

```diff
@@ -928,6 +942,110 @@ def sweep(self, start, stop, step=None, num=None):
                                 step=step, num=num)
 
 
+class ParameterWithSetpoints(Parameter):
```

Comment:
`Simple Example of ParameterWithSetpoints` notebook is great! (modulo some typos, but ok)